Roslyakov Evgeny/oop.js

```diff
@@ -0,0 +1,266 @@
+//---------- Class MenuItem ----------//
+/*@constructor
+ *@param name       item name
+ *@param price      item price
+ *@param calories   item calories number
+ */
+function MenuItem(options) {
+    this.name = options.name;
+    this.price = options.price;
+    this.calories = options.calories;
+}
+
+/* Get the item name */
+MenuItem.prototype.getName = function () {
+    if(this.name === undefined) {
+        return this.size.name;
```

Comment:
Btw, are you sure that you can use this in a parent class? It doesn't have property `size`, so if you try to use Salad.prototype.getName() you will receive an Error Cannot read property 'name' of undefined

Reply:
Yes, salad won't pass this condition 'cause it doesn't have a `size` property. But it will return this.name in this case. Overall, this getName method returns `size` in the hamburger's case and `name` in the others.